addons/account/models/account_move.py

```diff
@@ -4254,40 +4254,35 @@ def get_account_notification(partner, moves, is_success):
             ]
 
         limit = job_count + 1
-        to_process = self.env['account.move']._read_group(
+        to_process = self.env['account.move'].search(
             [('send_and_print_values', '!=', False)],
-            groupby=['company_id'],
-            aggregates=['id:recordset'],
             limit=limit,
+            order='id asc',
         )
         need_retrigger = len(to_process) > job_count
         if not to_process:
             return
 
-        for _company, moves in to_process[:job_count]:
+        all_moves = to_process[:job_count]
+        for _company, moves in all_moves.grouped('company_id').items():
             try:
                 # Lock moves
                 with self.env.cr.savepoint(flush=False):
                     self._cr.execute('SELECT * FROM account_move WHERE id IN %s FOR UPDATE NOWAIT', [tuple(moves.ids)])
 
-            except OperationalError as e:
-                if e.pgcode == '55P03':
-                    _logger.debug('Another transaction already locked documents rows. Cannot process documents.')
-                else:
-                    raise
+            except psycopg2.errors.LockNotAvailable:
```

Comment:
I'm surprised we don't use OperationalError here because we are using that in a lot of places in odoo for the same reason :thinking:

Reply:
LockNotAvailable is a subclass of OperationalError, it's just another way to write the same thing, which I find more readable but it's fine by me if you want me to rollback that part to the previous state :slightly_smiling_face: